_posts/2022-11-28-swig-for-c-and-lua.md

```diff
@@ -0,0 +1,491 @@
+---
+title: "Using SWIG to generate bindings between C and Lua"
+description: Trying out SWIG to automate generation of binding layer between Lua and C worlds
+author: stawiski
+---
+
+<!-- excerpt start -->
+
+Integrating Lua and C codebases requires adding a lot of boilerplate code. Lua must understand C data structures and functions in its own way to handle them and vice versa. That’s a lot of wrapper code integrating two languages together that must be written and then maintained as the application evolves. In this article, we will explore SWIG's promise of automating this code generation.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## What is SWIG?
+
+[SWIG](https://www.swig.org/) is a tool that generates a binding layer between C/C++ and higher-level programming languages. The promise of SWIG is simple: write an interface file that annotates C/C++ headers with custom SWIG markers, and generate bindings to a given language, so that you can expose whatever functionality you want from C/C++ and use it in that language.
```

Comment:
```suggestion
[SWIG](https://www.swig.org/) is a tool that generates a binding layer between C/C++ and higher-level programming languages. The promise of SWIG is simple: write an interface file that annotates C/C++ headers with custom SWIG markers, and generates bindings to a given language, so that you can expose whatever functionality you want from C/C++ and use it in that language.
```